packages/webapi/api.go

```
@@ -33,8 +33,21 @@ import (
 
 const APIVersion = 1
 
-func addHealthEndpoint(server echoswagger.ApiRoot) {
-	server.GET("/health", func(c echo.Context) error { return c.NoContent(http.StatusOK) }).
+func AddHealthEndpoint(server echoswagger.ApiRoot, chainService interfaces.ChainService, metricsService interfaces.MetricsService) {
+	server.GET("/health", func(e echo.Context) error {
+		chainIDs, err := chainService.GetAllChainIDs()
+		if err != nil {
+			return e.String(http.StatusInternalServerError, fmt.Sprintf("failed to get all chain IDs: %s", err))
```

it's safe to panic in case of an internal error, it should be cached and logged by the middleware

---

Um. Actually this part I have a question. We need to return an error if the node is unhealthy, but what is the error that we are going to return?